RELEASES.md

```diff
@@ -205,3 +205,95 @@ To create a minor release:
 6. Create a pull request back to master with the CHANGELOG & version changes from the latest release.
    - Remove all `R:minor` labels from the pull requests that were included in the release.
    - Do not merge the backport branch into master.
+
+
+## Minor Release Checklist
+
+The following set of steps are performed on all releases that increment
+the _minor_ version. These steps ensure that Tendermint is well tested, stable,
+and suitable for adoption by the various diverse projects that rely on Tendermint.
+
+### Feature Freeze
+
+Ahead of any minor version release of Tendermint, the software enters a two
+week 'Feature Freeze'. A feature freeze means that _no_ new features are added
+to the code base. The Tendermint team instead directs all attention to ensuring
+that the existing code is stable and reliable. Broken tests are fixed,
```

Comment:
I would further say that any changes that are not directly related to the features to be shipped (bug fixes, new tests, etc.) should not be committed, to avoid devoting an important testing effort to a moving target.
Of course, high severity bugs, security fixes, fixes to previously shipped features, should still be committed.

Reply:
> Dependency updates are the one of these items that we might want to omit from the list.

I don't really agree with this. I think we need a thorough discussion on this one.

If we devote a sizeable amount on time and effort at doing serious pre-release testing, we shouldn't be changing our systems on release branches unless it is for a good reason. We can't automatically be integrating dependency updates just because dependabot proposed it. When you change code that has been thoroughly tested, you are introducing risk. So at least we need to be convinced that the benefits outweigh the risks.

If we see that a dependency update is bringing a security fix, a serious bugfix that affects us, etc. then, I'm all for updating a release branch. But in that case, some effort has to be put (and planned for) into assessing the risks and benefits. Yes, it is not cheap, but I think it's a price worth paying for a software on which the entire Cosmos ecosystem depends.

Reply:
> If we see that a dependency update is bringing a security fix, a serious bugfix that affects us, etc. then, I'm all for updating a release branch.

I definitely have to agree with @sergio-mena on this. We should treat dependabot's dependency updates as updates that we need to critically review, just like all other PRs.

In addition to the stability concerns that @sergio-mena raised, I see Go as the least likely ecosystem (as compared to JavaScript, Rust, etc.) to be affected by supply chain attacks _iff_ we review dependency updates a little more carefully (of course, it's a tradeoff there in terms of how much energy we're willing to spend doing such reviews vs working on new features/fixing bugs).

Otherwise we're just as susceptible to supply chain attacks as the JavaScript ecosystem.

Beyond this, I personally feel that we should be endeavoring to reduce the number of dependencies to the bare minimum in Tendermint over time anyways.

Reply:
I think that we should avoid merging dep changes during our code freezes, however, at the moment and with the current staffing level, thoroughly reviewing each dependency is frankly beyond the capacity of the team. Additionally, this is some what of a different conversation to the conversation of the release process that should probably be discussed elsewhere.

Reply:
@sergio-mena I've added discussion to this document to elaborate on what you suggested, please take a look.

Reply:
Thanks @williambanfield, I read through the code and I 100% agree with it.
About reviewing dependencies, I think I have the same perspective: either we have the bandwidth to review them (and incorporate them into a code-frozen branch), or we don't merge them onto that branch (we can always merge to "master").